analysis/SConstruct

```diff
@@ -33,32 +30,35 @@ SourceFileScanner.add_scanner('.lyx', Scanner(gs.misc.lyx_scan, recursive = True
 # Load Stata executable if Stata builder is defined
 if 'BuildStata' in env['BUILDERS'].keys():
     env['stata_executable'] = gs.misc.load_yaml_value('config_user.yaml', 'stata_executable')
-
 # Export environment
 Export('env')
-
-# Run sub-trees
-SConscript('source/prepare_data/SConscript')
-SConscript('source/descriptive/SConscript')
-Default('./build', './release')
-
 # Additional mode options
 if mode == 'cache':
     CacheDir(cache_dir)
 
-# Debrief args is a dictionary storing the arguments that are useful for functions to run 
-# after all Scons targets have been built. Scons_debrief will pass these args into those functions. 
-debrief_args = {
-               'look_in'            : 'release;source',    # Folders to look in for large versioned files
-               'file_MB_limit_lfs'  : 2,                   # Soft limit on file size (w/ LFS);
-               'total_MB_limit_lfs' : 500,                 # Soft limit on total size (w/ LFS);  
-               'file_MB_limit'      : 0.5,                 # Soft limit on file size (w/o LFS);
-               'total_MB_limit'     : 125,                 # Soft limit on total size (w/o LFS);  
-               'lfs_required'       : gs.misc.load_yaml_value('config_global.yaml', 'prereq_git-lfs'), # Check if the repo requires LFS
-               'git_attrib_path'    : '../.gitattributes', # .gitattributes location
-               'MAXIT'              : 10,                  # max number of iterations to write file information to log
-               'log'                : 'state_of_repo.log'  # state of repo log file location
-               }
+# Logging (except on dry run)
+if not gs.misc.is_scons_dry_run(cl_args_list = sys.argv):
```

Comment:
I don't think this cuts it, @arosenbe . This `if` handling should be done within gslab_scons, not outsourced to the SConstruct. This adds clutter and logic to the SConstruct in a way that I think is undesirable (sufficiently so to keep things at status quo).

I can attempt to come up with a solution that have this in gslab_python if you want.

Reply:
I disagree, but am happy to discuss more/hear your idea. 

The other method I thought of was that that all our logging functions could accept a keyword argument with information about the dry-run status. Then the loggers toggle on or off depending on the argument's value. I think something like this is actually more confusing because it's less obvious when the logger is actually turned on. Hard to be more transparent about `condition()` than `if condition()`.

It also doesn't save much code since we'll still need something like this line to determine the dry-run status in the SConstruct. (Note that this is the only new line in the entire SConstruct, everything else is a reorganization.)

Reply:
The sense that I get is that MG prefers to keep as much "stuff" out of the SConstruct as possible, and this goes against it (adding an indent and a logical if block).

Reply:
I would prefer if the `start_log` and `end_log` functions read in `sys.argv` and bypasses itself when `--dry-run` is included.

Reply:
So you're voting to move the if switch into the logging functions? Easy to do.

Reply:
Yup! That's what I meant by "moving the if handling to gslab_scons, not keep it in the SConstruct." Sorry I wasn't clearer.

Reply:
You were very clear. I just wanted *even* more confirmation. 

While I have your attention fixed on logging. How do you feel about the current location of the logging function for input directories?

Reply:
I would prefer them in gslab_scons.misc actually --- I just noticed that you changed it to configuration.py in this PR --- what's the rationale?

Reply:
Nope, it was always in configuration.py. It used to run as part of the `configuration` function, but that makes some of the argument passing redundant when we try to skip the logging process. (I put all the logging calls in one location).

I think we thought it was too template-y for inclusion in gslab_python, but I see an easy way to generalize it.

Reply:
On second thought, it does seem quite "template-y". I'm happy to keep it where it is.

Reply:
I'd be happy to move it. A dream of mine would be to make our logging functions easy to find: put `log`, `scons_debrief`, and `record_dir` in their own files. 

I would rewrite the function to accept any dictionary and a key. Basically rename `PATHS` to `d` and `input_assets_key` to `record_key`.